app/config/collections.php

```diff
@@ -3793,6 +4078,564 @@
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => []
+            ],
+            [
+                '$id' => ID::custom('providerBranch'),
+                'type' => Database::VAR_STRING,
+                'format' => '',
+                'size' => Database::LENGTH_KEY,
+                'signed' => true,
+                'required' => true,
+                'default' => null,
+                'array' => false,
+                'filters' => []
+            ],
+        ],
+        'indexes' => [
+            [
+                '$id' => ID::custom('_key_installationId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['installationId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_installationInternalId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['installationInternalId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_projectInternalId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['projectInternalId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_projectId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['projectId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_providerRepositoryId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['providerRepositoryId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_providerPullRequestId'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['providerPullRequestId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+            [
+                '$id' => ID::custom('_key_providerBranch'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['providerBranch'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ],
+        ],
+    ],
+
+    'vcsCommentLocks' => [
+        '$collection' => ID::custom(Database::METADATA),
+        '$id' => ID::custom('vcsCommentLocks'),
+        'name' => 'vcsCommentLocks',
+        'attributes' => [],
+        'indexes' => []
+    ],
```

Comment:
Why does this need to be a separate collection ? Can the race condition not be solved by having a `lock` attribute in the `vcsComments` collection ?

Reply:
Probably, Ill take a look. Initial idea was to have separate lock table to potentially have generic locking mechanism in future.